lib/visitor.js

```diff
@@ -389,10 +389,12 @@ class Visitor {
         const returnType = operation.returns
             ? this.resolver.resolveAndRequire(operation.returns, file)
             : { typeName: 'void', typeInfo: { isArray: false, inflection: { singular: 'void', plural: 'void' } } }
-        const returns = this.inlineDeclarationResolver.getPropertyDatatype(
+        let returns = this.inlineDeclarationResolver.getPropertyDatatype(
             returnType,
             returnType.typeInfo.isArray ? returnType.typeName : returnType.typeInfo.inflection.singular
         )
+        // results of operations are effectively a Promise at runtime
+        returns = operation.returns ? `Promise<${returns}>` : returns
```

Comment:
@daogrady To make operations' return types work, I had to wrap the result type into a `Promise`.  This seems to reflects the runtime behavior.  They always return promises, even if their impl. is non-async.  The [docs](https://cap.cloud.sap/docs/guides/providing-services#calling-actions-functions) indicate the same, although we might want to cross-check w/ the runtime folks.

Reply:
@sjvans maybe you can comment on return values of actions/functions. Always Promises?

Reply:
Sure! I have moved the promisification to a [module with similar functions](https://github.com/cap-js/cds-typer/blob/main/lib/components/wrappers.js). Did you check if that had any implications on handler implementation? I.e.


```cds
// MyService.cds
namespace my.namespace;

service MyService {
    action myUnboundAction returns Integer;
}
```

```js
// MyService.js|ts
const { myUnboundAction } = require('#cds-models/MyService')

// ...

this.on(myUnboundAction, () => {
  return 42   // will the type system now bug the user about 42 not being a Promise?
})
```

Reply:
Good point.  Indeed, promises are only accepted if the handler is async.  If it is sync, it gets rejected - at least it should be rejected, but is 'catched' by a different `on` overload that has an `any` return type.  A topic for a different day, I'd say.

I have changed the type to `Promise<return> | return`.